x-pack/plugin/mapper-counted-keyword/src/main/java/org/elasticsearch/xpack/countedkeyword/CountedKeywordFieldMapper.java

```diff
@@ -348,20 +427,27 @@ protected void parseCreateField(DocumentParserContext context) throws IOExceptio
             context.doc().add(new KeywordFieldMapper.KeywordField(fullPath(), new BytesRef(value.getKey()), fieldType));
             counts[i++] = value.getValue();
         }
-        BytesStreamOutput streamOutput = new BytesStreamOutput();
-        streamOutput.writeVIntArray(counts);
-        context.doc().add(new BinaryDocValuesField(countFieldMapper.fullPath(), streamOutput.bytes().toBytesRef()));
+        if (counts.length > 0) {
```

Comment:
+1 this should work. If count array is empty than there should also be no values. And assumption in `CountedKeywordSortedBinaryDocValues` that if there is a value then there should be a entry in count field should not break.

Maybe instead we can even stop earlier by adding this statement before `I` variable is initialized:

```java
if (values.isEmpty()) {
   return;
}
```